Library/Statements/TryCatchStatement.php

```diff
@@ -85,7 +89,11 @@ public function compile(CompilationContext $compilationContext)
                     $variable = $compilationContext->symbolTable->getTempVariableForWrite('variable', $compilationContext, $compilationContext);
                 }
 
-                $compilationContext->backend->copyOnWrite($variable, 'EG(exception)', $compilationContext);
+                if ($compilationContext->backend->isZE3()) {
+                    $assignExceptionVarStmt = $exprBuilder->statements()->rawC('ZEPHIR_CPY_WRT(&' . $variable->getName() . ', &' . $exc_var->getName() . ');');
+                } else {
+                    $assignExceptionVarStmt = $exprBuilder->statements()->rawC('ZEPHIR_CPY_WRT(' . $variable->getName() . ', ' . $exc_var->getName() . ');');
+                }
 
```

Comment:
The above is a dirty hack, but I failed to make this:

```php
                                $exprBuilder->statements()->let(array(
                                    $exprBuilder->operators()->assignVariable($variable->getName(), $exprBuilder->variable($variable->getName()))
                                ))->build(),
```

or similar constructs work :-(

If someone knows how to make them work, I would prefer it to this code.